compute/disks/create_regional_from_snapshot.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_regional_disk_create_from_snapshot]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// createRegionalDiskFromSnapshot creates a new regional disk with the contents of
+// an already existitng zonal disk snapshot.
+func createRegionalDiskFromSnapshot(
+	w io.Writer,
+	projectID, region string, replicaZones []string,
+	diskName, diskType, snapshotLink string,
+	diskSizeGb int64,
+) error {
+	// projectID := "your_project_id"
+	// region := "us-west3"
+	// replicaZones := []string{"us-west3-a", "us-west3-b"}
+	// diskName := "your_disk_name"
+	// diskType := "regions/us-west3/diskTypes/pd-ssd"
+	// snapshotLink := "projects/your_project_id/global/snapshots/snapshot_name"
+	// diskSizeGb := 120
+
+	// Exactly two replica zones must be specified
+	replicaZoneURLs := []string{
+		fmt.Sprintf("projects/%s/zones/%s", projectID, replicaZones[0]),
+		fmt.Sprintf("projects/%s/zones/%s", projectID, replicaZones[1]),
+	}
+
+	ctx := context.Background()
+	disksClient, err := compute.NewRegionDisksRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewRegionDisksRESTClient: %v", err)
```

Comment:
NIT: The current best practice is to use `%w` when passing errors along.

Reply:
All of these samples use `%v`, as do the rest of compute samples. How many of them do you want me to refactor? All disk samples, all compute samples, all Go samples? Do you want this to happen in this PR?

Reply:
Let's switch over to using `%w` starting with this PR. We can refactor other samples on a "best efforts" basis.